crates/router/src/core/payment_methods/cards.rs

```diff
@@ -3031,6 +3032,15 @@ pub async fn list_payment_methods(
                             .connector
                             .connector_name
                             .to_string()
+                        && first_routable_connector
+                            .connector
+                            .merchant_connector_id
+                            .as_ref()
+                            .map(|merchant_connector_id| {
+                                *merchant_connector_id.get_string_repr()
+                                    == intermediate.merchant_connector_id
+                            })
+                            .unwrap_or_default()
```

Comment:
If mca id is not present, u are making it false, is it expected?

Reply:
Yes, mca_id not present will be a case that would never occur, and even if it occurs we wouldn't want to retain the connector